### PR TITLE
Fix demi bold font

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

This was really rough.

TL;DR "demi" is not a valid weight suffix, but "demibold" is
https://github.com/facebook/react-native/blob/main/React/Views/RCTFont.mm#L23

Discussion:
https://github.com/facebook/react-native/issues/20788

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Locally w/ sign up flow

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

**Review by sliptype — APPROVED** [review]
Interesting

I think we should add DemiBold to the `Text` component as well
